### ISSUE TITLE
How to keep some sort of ordering between joined streams?

### ISSUE THREAD

**Issue by limx0** [mrocklin/streamz#155]
Say I have two source streams on kafka, one updating at 1 message per second, and one updating at 5 messages per second. 

What would be the most appropriate way to not allow the streams to get more than ~10s out of sync? I effectively want to push

**Comment by CJ-Wright** [comment]
Is there some code which says that the source died? If so then you could maybe push all the data into a buffer while waiting for the source to come back.

Similarly with the first scenario, you could have a conditional which depending on the time out of sync either pushes data through the normal route, or to a buffer until new data comes in.

**Comment by limx0** [comment]
Yeah I like your second idea @CJ-Wright.  I will whip something up - I wonder whether others might find it useful enough for a PR

**Comment by CJ-Wright** [comment]
I've used a combination of joining nodes, filter, and pluck to essentially gate certain paths. Maybe to gates would work for pushing data directly into the processing or into a buffer.
